### ISSUE TITLE
Usage on NixOS

### ISSUE THREAD

**Issue by MikulasVanousek** [tadfisher/android-nixpkgs#29]
I use NixOS and would like to set up the Android environment globally. I managed to start a nix shell with the environment using the provided file. I did not figure out, how to add it to the `configuration.nix`. Is there a way to import the shell file into the configuration? Or is there a way to adapt the code, so it is possible to paste it into a config file?

**Comment by MikulasVanousek** [comment]
I cracked it.

**Comment by MikulasVanousek** [comment]
This is the `android.nix` config file I came up with to add android SDK to global environment. 
```

{ config, pkgs, ... }:

let
  android-nixpkgs = pkgs.callPackage <android-nixpkgs> {
    # Default; can also choose "beta", "preview", or "canary".
    channel = "stable";
  };
android-sdk = android-nixpkgs.sdk (sdkPkgs: with sdkPkgs; [
    cmdline-tools-latest
    build-tools-32-0-0
    platform-tools
    platforms-android-31
    emulator
  ]);
in {
  environment.systemPackages = with pkgs; [
    # IDE
    android-tools
    android-sdk
    android-studio
    flutter
  ];
}
```
This seems to install the packages, but the `ANDROID_HOME` and `ANDROID_SDK_ROOT` `variables remain unset. Do you have any idea why this could be?

**Comment by MikulasVanousek** [comment]
If I run the file you provided with `nix-shell`, it does set the required variables.

**Comment by tadfisher** [comment]
Yes, I personally use this for an isolated Android dev environment via flakes. I used to use this via home-manager as well, but it makes less sense to do so with flakes support.

The `sdk` derivation includes `setupHook` and `shellHook` to set those environment variables, but that won't be picked up by NixOS via `environment.systemPackages`. You'll need to set those yourself, e.g.

```nix
environment.variables = {
  ANDROID_SDK_ROOT = "${android-sdk}/share/android-sdk";
  ANDROID_HOME = "$ANDROID_SDK_ROOT";
};
```

**Comment by MikulasVanousek** [comment]
This is my current `android.nix`:
```
{ config, pkgs, ... }:

let
  android-nixpkgs = pkgs.callPackage <android-nixpkgs> {
    # Default; can also choose "beta", "preview", or "canary".
    channel = "stable";
  };
android-sdk = android-nixpkgs.sdk (sdkPkgs: with sdkPkgs; [
    cmdline-tools-latest
    build-tools-32-0-0
    platform-tools
    platforms-android-31
    emulator
  ]);
in {
  environment.systemPackages = with pkgs; [
    # IDE
    android-tools
    android-sdk
    android-studio
    flutter
  ];
  environment.variables = {
    ANDROID_SDK_ROOT = "${android-sdk}/share/android-sdk";
    ANDROID_HOME = "$ANDROID_SDK_ROOT";
  };
}
```
I think it would be a good idea to add NixOS specific instructions to the README.md, but first, is there a way to add the channel decoratively? The git bootstrapping did not seem to work, and I didn't find it in the manual.

**Comment by MikulasVanousek** [comment]
> Yes, I personally use this for an isolated Android dev environment via flakes. I used to use this via home-manager as well, but it makes less sense to do so with flakes support.
> 
> The `sdk` derivation includes `setupHook` and `shellHook` to set those environment variables, but that won't be picked up by NixOS via `environment.systemPackages`. You'll need to set those yourself, e.g.
> 
> ```nix
> environment.variables = {
>   ANDROID_SDK_ROOT = "${android-sdk}/share/android-sdk";
>   ANDROID_HOME = "$ANDROID_SDK_ROOT";
> };
> ```

Interestingly enough, this does not actually set `ANDROID_HOME`. I am not exactly sure why, but I think the `$` might be misinterpreted by nix. I used:
```
environment.variables = let home= "${android-sdk}/share/android-sdk"; in {
    ANDROID_SDK_ROOT = home;
    ANDROID_HOME = home;
  };
```
Instead. 

Flutter now picks up on the SDK -- Hurray! But accepting the licenses does not work for me. 
```
❯ flutter doctor --android-licenses
Error: Could not determine SDK root.
Error: Either specify it explicitly with --sdk_root= or move this package into its expected location: <sdk>/cmdline-tools/latest/
```
Is it perhaps because of the read-only file system? Did accepting licenses work for you?

**Comment by purepani** [comment]
> > Yes, I personally use this for an isolated Android dev environment via flakes. I used to use this via home-manager as well, but it makes less sense to do so with flakes support.
> > The `sdk` derivation includes `setupHook` and `shellHook` to set those environment variables, but that won't be picked up by NixOS via `environment.systemPackages`. You'll need to set those yourself, e.g.
> > ```nix
> > environment.variables = {
> >   ANDROID_SDK_ROOT = "${android-sdk}/share/android-sdk";
> >   ANDROID_HOME = "$ANDROID_SDK_ROOT";
> > };
> > ```
> 
> Interestingly enough, this does not actually set `ANDROID_HOME`. I am not exactly sure why, but I think the `$` might be misinterpreted by nix. I used:
> 
> ```
> environment.variables = let home= "${android-sdk}/share/android-sdk"; in {
>     ANDROID_SDK_ROOT = home;
>     ANDROID_HOME = home;
>   };
> ```
> 
> Instead.
> 
> Flutter now picks up on the SDK -- Hurray! But accepting the licenses does not work for me.
> 
> ```
> ❯ flutter doctor --android-licenses
> Error: Could not determine SDK root.
> Error: Either specify it explicitly with --sdk_root= or move this package into its expected location: <sdk>/cmdline-tools/latest/
> ```
> 
> Is it perhaps because of the read-only file system? Did accepting licenses work for you?

I am also seeming to have this issue. I have tried manually setting the android_sdk configuration in flutter with the same issue.

**Comment by FlafyDev** [comment]
> > Instead.
> > Flutter now picks up on the SDK -- Hurray! But accepting the licenses does not work for me.
> > ```
> > ❯ flutter doctor --android-licenses
> > Error: Could not determine SDK root.
> > Error: Either specify it explicitly with --sdk_root= or move this package into its expected location: <sdk>/cmdline-tools/latest/
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is it perhaps because of the read-only file system? Did accepting licenses work for you?
> 
> I am also seeming to have this issue. I have tried manually setting the android_sdk configuration in flutter with the same error. I am using a flake however, based on the template given.

Currently having the same issue with Flutter and also using the flake's template

**Comment by MikiVanousek** [comment]
This is still an issue for me.
